fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkimport/ChunkWriter.java

```diff
@@ -108,32 +121,98 @@ public void writeItems(List<java.lang.Object> arg0) throws Exception {
             fhirDatastoreId = Constants.DEFAULT_FHIR_TENANT;
             logger.info("writeItems: Set DatastoreId to default!");
         }
-        cosClient = COSUtils.getCosClient(cosCredentialIbm, cosApiKeyProperty, cosSrvinstId, cosEndpintUrl,
-                cosLocation);
-        if (cosClient == null) {
-            logger.warning("writeItems: Failed to get CosClient!");
-            return;
-        } else {
-            logger.finer("writeItems: Got CosClient successfully!");
-        }
 
         FHIRConfiguration.setConfigHome("./");
         FHIRRequestContext.set(new FHIRRequestContext(fhirTenant, fhirDatastoreId));
 
         FHIRPersistenceHelper fhirPersistenceHelper = new FHIRPersistenceHelper();
         FHIRPersistence fhirPersistence = fhirPersistenceHelper.getFHIRPersistenceImplementation();
         FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null);
+        FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
 
-        int imported = 0;
+        int processedNum = 0, succeededNum =0, failedNum = 0;
+        ImportTransientUserData chunkData = (ImportTransientUserData) stepCtx.getTransientUserData();
         for (Object objResJasonList : arg0) {
-            List<String> resCosObjectNameList = (List<String>) objResJasonList;
+            List<Resource> fhirResourceList = (List<Resource>) objResJasonList;
+
+            for (Resource fhirResource : fhirResourceList) {
+                try {
```

Comment:
It's possible move this code inside `for` to other method.

Reply:
I don't think add unnecessary method is a good idea even though the cost of function call could be minor, but this can be a huge loop. for function call, there will always be stack operations there.